src/main/java/org/apache/flink/benchmark/MultiInputCheckpointingTimeBenchmark.java

```diff
@@ -0,0 +1,189 @@
+                context.miniCluster.triggerCheckpoint(context.jobID);
+        checkpoint.get();
+    }
+
+    @State(Scope.Thread)
+    public static class DebloatedCheckpointEnvironmentContext extends FlinkEnvironmentContext {
+        public JobID jobID;
+
+        @Override
+        public void setUp() throws Exception {
+            super.setUp();
+            env.setParallelism(JOB_PARALLELISM);
+            env.enableCheckpointing(Long.MAX_VALUE);
+
+            DataStream<Record> source1 =
+                    env.fromSource(
+                                    new RecordSource(
+                                            Integer.MAX_VALUE, (int) SMALL_RECORD_SIZE.getBytes()),
+                                    noWatermarks(),
+                                    RecordSource.class.getName())
+                            .slotSharingGroup("source-small-records")
+                            .rebalance();
+
+            DataStream<Record> source2 =
+                    env.fromSource(
+                                    new RecordSource(
+                                            Integer.MAX_VALUE, (int) BIG_RECORD_SIZE.getBytes()),
+                                    noWatermarks(),
+                                    RecordSource.class.getName())
+                            .slotSharingGroup("source-big-records")
+                            .rebalance();
+
+            source1.connect(source2)
+                    .map(
+                            new CoMapFunction<Record, Record, Record>() {
+                                @Override
+                                public Record map1(Record record) throws Exception {
+                                    return record;
+                                }
+
+                                @Override
+                                public Record map2(Record record) throws Exception {
+                                    return record;
+                                }
+                            })
+                    .name("co-map")
+                    .slotSharingGroup("map-and-sink")
+                    .addSink(new SlowDiscardSink<>())
+                    .slotSharingGroup("map-and-sink");
+
+            final StreamGraph streamGraph = env.getStreamGraph(false);
+            final JobGraph jobGraph = streamGraph.getJobGraph();
+            final List<JobVertex> vertices = jobGraph.getVerticesSortedTopologicallyFromSources();
+
+            final JobClient jobClient = env.executeAsync();
+            jobID = jobClient.getJobID();
+            CommonTestUtils.waitForAllTaskRunning(miniCluster, jobID, false);
+            List<JobVertexID> sourceIds =
+                    Arrays.asList(vertices.get(0).getID(), vertices.get(1).getID());
+            BackpressureUtils.waitForBackpressure(
+                    jobID, sourceIds, miniCluster.getRestAddress().get());
+            Thread.sleep(DEBLOATING_STABILIZATION_PERIOD);
```

Comment:
This I think we could also deduplicate quite easily in the base class?

Reply:
I understand what you mean. If we happen to need to tweak the parameters differently in those two benchmark, I think we should just then have such parameters defined separately in two files. I think we can achieve this quite easily via some abstract getters in the context class?

However if they are now the same, I would like to avoid a situation when someone in the future changes a parameter only in one file, but he simply forgets to change/check it in the other one, causing lot's of confusion for next one in line that will notice the unnecessary difference.

What I'm trying to say, is that ideally we should come up with a code structure that reflects both of those aspects - a shared parameters and private ones (if we need them), which would give us more self documenting code.